azure/Guidelines.md

```diff
@@ -0,0 +1,310 @@
+
+The OneAPI guidelines are available on Github at [https://github.com/microsoft/api-guidelines][1].
+
+Highlights of the OneAPI guidelines include:
+
+* **REST fundamentals**.  Guidance around REST basics such as HTTP headers, verbs, status codes, and other similar areas.
+* **Versioning**.  The guidance for versioning covers the externally facing mechanics of versioning and compatibility guarantees between versions.  Guidance for when API owners should increment versions is also present.
+* **CORS**.  Guidance around the use of CORS.
+* **Authentication**.  Guidance around the authentication and its use.
+* **Encoding**.  Guidance around the use of JSON encoding and standards.
+* **Error responses**.  A standard format for error responses is closed.
+* **Dates and times**.  Detailed guidance around date and time formats and encoding is provided.
+* **Relationship to OData**.  Guidance around the relationship between OData and REST APIs.
+* **JSON representation**.  JSON is preferred over XML and other formats.
+* **Long running operations**.  The OneAPI guidelines are updated and simplified over the previous Azure guidelines.
+* **Push notifications via Webhooks**.
+
+### Updating from previous versions of the Azure API guidelines
+
+The OneAPI guidelines started from the last revision of the Azure API Guidelines and are highly compatible. Azure services that are already using an earlier version of the guidelines should not require much additional work.
+
+#### Asynchronous operations
+
+The OneAPI guidelines for Long Running Operations guidelines are an updated, clarified and simplified version of the Asynchronous Operations guidelines from the 2.1 version of the Azure API guidelines. Unfortunately, to generalize to the whole of Microsoft and not just Azure, the board decided to rename the HEADER used in the operation from `Azure-AsyncOperation` to `Operation-Location`. Services can and **SHOULD** support both `Azure-AsyncOperation` and `Operation-Location` HEADERS, even though they are redundant so that existing SDKs and clients will continue to operate. Clients that call these services **SHOULD** look for both HEADERS and prefer the `Operation-Location` version. Both HEADERS **MUST** return the same value.
+
+### Additional guidance for Azure Resource Manager resource providers
+
+Teams building ARM RPs MUST follow the additional guidance in the ARM RPC and related documents. These documents can be found here.
+
+* [Azure Resource Manager Documents][2] (Internal only)
```

Comment:
I think these are public on Github now

Reply:
I'll dig - if anyone happens to know where, let me know.  I'd rather include public docs.

Reply:
https://github.com/Azure/azure-resource-manager-rpc